### PR TITLE
fix(query): fix mysql handler overflow

### LINKED ISSUE #9236: bug: thread 'mysql-query-executor' has overflowed its stack
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

8.0.26-v0.8.150-nightly

### What's Wrong?

query.log
```
thread 'mysql-query-executor' has overflowed its stack
fatal runtime error: stack overflow
```

### How to Reproduce?

```sql
DROP DATABASE IF EXISTS databend0;
CREATE DATABASE databend0;
USE databend0;
CREATE TABLE t0(c0INT BIGINT NULL, c1FLOAT DOUBLE NULL DEFAULT(0.410796195268631)); -- 57ms;
CREATE TABLE t1(c0INT INT8 NULL DEFAULT(-1949367821)); -- 40ms;
CREATE TABLE t2(c0FLOAT DOUBLE NULL); -- 49ms;

SELECT t2.c0float, t0.c0int, t0.c1float, t1.c0int FROM t1, t2 LEFT  JOIN t0 ON (((- ((-116704857)+(t0.c0int))))<((+ ((t0.c0int)*(t0.c0int))))) WHERE (((((((0.11264920979738235 NOT BETWEEN 0.9119921326637268 AND 0.8280299305915833))AND((true IN (true, ((false)AND(false)), (t0.c1float NOT BETWEEN t0.c1float AND 0.8534555435180664))))))and((((('')LIKE('hYAnCXx')))AND(((((((false)or(false)))or(false)))or(true)))))))OR((NOT ((NULL)=(NULL)))))
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by ClSlaid**
Signed-off-by: 蔡略 <cailue@bupt.edu.cn>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

make some optimizer functions iterative

Closes #9236

### CODE REVIEW

File: src/query/sql/src/planner/optimizer/rule/rewrite/rule_normalize_disjunctive_filter.rs [hunk 1]
```diff
@@ -170,50 +174,147 @@ impl Rule for RuleNormalizeDisjunctiveFilter {
     }
 }
 
+// rewrite in post order traversal
```
Comment (Xuanwo) [inline src/query/sql/src/planner/optimizer/rule/rewrite/rule_normalize_disjunctive_filter.rs:177]: Hi, can you add a unit test or SQL logic test for this change?

**Comment by ClSlaid** [comment]
This PR currently does not solve the issue, I'll find out the real cause and add a test case in the SQL logic test.

**Comment by ClSlaid** [comment]
This PR currently still not solve the problem, but changes how it goes. Instead of a stack overflow, the heap will grow slowly and steadily. This may be caused by a potential logic problem in our optimizer.

**Comment by ClSlaid** [comment]
closes for #9645

**Pull request closed without merging**